### DISCUSSION TITLE
how to choose the templates displayed in the submenu?

### DISCUSSION THREAD

**Discussion by twicemice** [TriliumNext/Trilium#4349] [Q&A]
hello again, 
I have created a series of templates to speed up my work. They all appear in the secondary menu when right clicked. All those with the #template tag appear. However, some appear repeated and also appear instances of template that I have used in some entries (with all the information), those whose attribute is ~template=
Is there any way to limit the templates that appear when using the secondary menu? How can I prevent them from appearing repeatedly? How can I prevent template instances from appearing in the secondary menu?
I am using the latest version of Trillium. 
Thanks

**Comment by Nriver** [comment]
Remove `#template` from your template note.

**Reply by zadam** [reply]
Here's the related documentation: https://github.com/zadam/trilium/wiki/Template

**Comment by twicemice** [comment]
Hi
Sorry, i think i haven´t explain my problem very well. What i´m trying to say is that in the secondary menu, clicking on the mouse right button, there are template options but there are instances of templates.
Imagine that i create a note with the template "customer" called "john doe". 
The template "customer" has the tag #template, the note "john doe" has the link "~template=customer". 
After that, when i try to create a new note with the secondary menu i have two options. The first one is the "customer" template, that creates a blank note with the elements of customer template. But the second option is called "customer" also, but if i choose this one, it generates a new note, with the elements of customers template, but filled with ther data from "john doe".  
This is very problematic because i have a long list of templates that gets bigger and bigger with the indivudual instances. 
Does anybody has the same problem?

**Reply by zerebos** [reply]
Did you make the `#template` attribute inheritable on the original template? That would make the attribute also appear on all instances of the template causing them to be picked up as templates and put in the menu.

**Comment by rodrigoalexandrecosta** [comment]
Can you post us some screenshots? I think that this a bug but not also that. Don't know if was reported somewhere but here it goes.

I suspect that perhaps your instance notes are inheriting the #template tag from your templates. Please check if your declaration of the template is something like "#template(inheritable)". In this case, even if the instance notes have the correct relation in the "Owned Attributes" ribbon, the #template label will propagate to the instances. Just remove the inheritance flag and you're good.

Now, the (possible) bug: I tried to reproduce that over here in one of my templates. The #template label actually do not appear in the "Inherited Attributes" ribbon, but the behavior is replicated, meaning that I am able to reproduce what @twicemice reported: not only my template note but also my instances notes from that template all appear in the "Insert note after" and "Insert child note" secondary mouse button menu in the tree.

I don't know exactly if this behavior would be useful for anyone, but anyway I think that the bug, if it's really a bug, is that the #template label does not show in the "Inherited Attributes" in the note ribbon.

**Reply by zadam** [reply]
So, the idea is that `#template` should not be inherited. The reason for this is that all attributes on template notes are always inherited to the instance notes, but inheriting `#template` like that would defeat the purpose of this label. It is pretty exceptional in this.

The bug was caused by separate implementation of attribute handling in search (used to find the templates) and the general attribute inheritance used to render the "Inherited attributes" ribbon. With the fix, the `#template` should be impossible to inherit, even if it has `isInheritable` flag.

**Comment by twicemice** [comment]
Hi,
Thank you all for uderstand my problem. 
@rauenzi @rodrigoalexandrecosta, I have reviewed the template notes and instance notes and i haven´t found inheritable atributes.
Sorry, i think it could be a bug.

**Reply by zerebos** [reply]
Can you provide some screenshots of the template and it's attributes?

**Reply by rodrigoalexandrecosta** [reply]
Also you should check the parents of your templates, since the relation can propagate
